ADR/0027-container-images.md

```diff
@@ -0,0 +1,98 @@
+# 27. Container Image Management Practice
+
+Date: 2023-06-30
+
+## Status
+
+Proposed
+
+* Relates to [ADR 17. Use our own pipelines](0017-use-our-pipelines.md)
+
+## Context
+
+The purpose of this document is to establish container image management practices for AppStudio container images that are deployed in the staging and production environments.  The goal is to ensure that AppStudio is continuously maintaining secure operations that are in accordance with the ESS SEC-PATCH-REQ-2 (OS Patching) requirements.
+
+### Scope
+* The scope of this process is limited to the images found in our [quay.io/organization/redhat-appstudio](https://quay.io/organization/redhat-appstudio) repository.
+* Images from dependencies that fall outside of this AppStudio process should follow the ESS SEC-PATCH-REQ-1 and ESS SEC-PATCH-REQ-2 processes.  It is up to the component teams to ensure they are adhering to these requirements.
+* Images that are not intended for the staging and/or production environments are out of scope.
+
+
+## Decision
+
+### Role
+
+**Component Team**: Develops and maintains components that are built as images and deployed as part of AppStudio
+
+
+### Responsibilities
+
+#### Automated Build and Scanning
+###### Onboard to Pipelines As Code (PaC)
+
+Component Teams are responsible for ensuring their container images are continuously built and scanned for vulnerabilities by following the
+[Extending the Service](https://redhat-appstudio.github.io/infra-deployments/docs/deployment/extending-the-service.html) process to onboard their component to the PaC service.
+
+***
+#### Container Images <br>
+
+###### Trigger Builds
+Under the PaC service, images are rebuilt when there are updates to the component’s git repository but additional configuration is needed in the Dockerfile to ensure the underlying base (UBI) images are updated with the latest packages (see the [HAS example](https://github.com/redhat-appstudio/application-service/blob/main/Dockerfile#L24])) or at the very least, the latest [security updates](https://developers.redhat.com/articles/2021/11/11/best-practices-building-images-pass-red-hat-container-certification#best_practice__5__include_the_latest_security_updates_in_your_image).   This will minimize the gap between patching and should meet our CVE timelines as long as the repository is active.
+
+Component teams are encouraged to install [renovatebot](https://github.com/renovatebot/renovate) to keep their dependencies up to date.
+
+###### Scheduled Builds
+
+Since image updates are based on how active our repos are, there is the risk that over time, as code stabilizes and/or enters maintenance mode, the triggers for rebuilds will be less frequent which will cause the images to degrade.  To avoid this, component teams should also ensure there are scheduled, weekly builds in place
```

Comment:
```suggestion
Since image updates are based on how active our repos are, there is the risk that over time, as code stabilizes and/or enters maintenance mode, the triggers for rebuilds will be less frequent which will cause the images to degrade.  To avoid this, component teams should also ensure there are scheduled, weekly builds or builds driven by renovatebot or dependabot in place.
```